packages/angular_devkit/build_angular/src/utils/index-file/inline-fonts.ts

```diff
@@ -0,0 +1,170 @@
+    if (hrefsContent.size === 0) {
+      return content;
+    }
+
+    // Replace link with style tag.
+    const { rewriter, transformedContent } = await htmlRewritingStream(content);
+    rewriter.on('startTag', tag => {
+      const { tagName, attrs } = tag;
+
+      if (tagName !== 'link') {
+        rewriter.emitStartTag(tag);
+
+        return;
+      }
+
+      const hrefAttr = attrs.some(({ name, value }) => name === 'rel' && value === 'stylesheet')
+        && attrs.find(({ name, value }) => name === 'href' && hrefsContent.has(value));
+      if (hrefAttr) {
+        const href = hrefAttr.value;
+        const cssContent = hrefsContent.get(href);
+        rewriter.emitRaw(`<style type="text/css" title="${href}">${cssContent}</style>`);
+      } else {
+        rewriter.emitStartTag(tag);
+      }
+    });
+
+    return transformedContent;
+  }
+
+  private async getResponse(url: string, userAgent: UserAgent): Promise<string> {
+    const key = `${packageVersion}|${url}|${userAgent}`;
+
+    if (cacheFontsPath) {
+      const entry = await cacache.get.info(cacheFontsPath, key);
+      if (entry) {
+        return promisify(readFile)(entry.path, 'utf8');
+      }
+    }
+
+    const data = await new Promise<string>((resolve, reject) => {
+      let rawResponse = '';
+      https.get(
+        url,
+        {
+          headers: {
+            'user-agent': userAgent,
+          },
+        },
+        res => {
+          res
+            .on('data', chunk => rawResponse += chunk)
+            .on('end', () => resolve(rawResponse.toString()));
+        },
+      )
+        .on('error', e => reject(e));
+    });
+
+    if (cacheFontsPath) {
+      await cacache.put(cacheFontsPath, key, data);
+    }
+
+    return data;
+  }
+
+  private async processHrefs(hrefList: string[]): Promise<Map<string, string>> {
+    const hrefsContent = new Map<string, string>();
+
+    for (const href of hrefList) {
+      // Normalize protocols to https://
+      let normalizedHref = href;
+      if (!href.startsWith('https://')) {
+        if (href.startsWith('//')) {
+          normalizedHref = 'https:' + href;
+        } else if (href.startsWith('http://')) {
+          normalizedHref = href.replace('http:', 'https:');
+        } else {
+          // Unsupported CSS href.
+          continue;
+        }
+      }
+
+      if (!SUPPORTED_PROVIDERS.some(url => normalizedHref.startsWith(url))) {
+        // Provider not supported.
+        continue;
+      }
+
+      // The order IE -> Chrome is important as otherwise Chrome will load woff1.
+      let cssContent = '';
+      if (this.options.WOFFSupportNeeded) {
+        cssContent += await this.getResponse(normalizedHref, UserAgent.IE);
+      }
+      cssContent += await this.getResponse(normalizedHref, UserAgent.Chrome);
+
+      if (this.options.minifyInlinedCSS) {
+        cssContent = cssContent
+          // Comments and new lines.
+          .replace(/(\n|\/\*\s.+\s\*\/)/g, '')
```

Comment:
nit: Can we factor out the two regexes and write some tests for them?
(address in a follow-up PR is fine)

Reply:
I actually left it separate because the new lines is rather simple. But I can split it.

There is a basic test for this here: https://github.com/angular/angular-cli/pull/18926/files#diff-3e96aa638f2afb73f78ff67569b9b185c46a32968faf2c85cf0c5e75987452d8R71-R80